### DISCUSSION TITLE
Can ZipFile open .7z files?

### DISCUSSION THREAD

**Discussion by wf-soft** [icsharpcode/SharpZipLib#784] [Q&A] [answered]
```
using (var zip = new ZipFile(@"D:\1.7z")) {
    Console.WriteLine(zip);
}
```
throw
"Cannot find central directory"

Since the .7z file could not be uploaded, I put it in the .zip file

[1.zip](https://github.com/icsharpcode/SharpZipLib/files/9771475/1.zip)

I try to use some other packages such as SharpCompress, which can be read normally.

**Comment by piksel** [comment, marked as answer]
`.7z` are not zip files, they are LZMA files created by 7-zip. SharpZipLib does not support 7-zip archives.
See https://en.wikipedia.org/wiki/7z

**Comment by wf-soft** [comment]
@piksel Ok. I successfully read a 7z, so maybe he renamed it from the zip file, and I thought he could support
